autopilot/contractor.go

```diff
@@ -662,11 +687,12 @@ func (c *contractor) runContractRenewals(ctx context.Context, w Worker, budget *
 	return renewed, nil
 }
 
-func (c *contractor) runContractRefreshes(ctx context.Context, w Worker, budget *types.Currency, renterAddress types.Address, toRefresh []contractInfo) ([]api.ContractMetadata, error) {
+func (c *contractor) runContractRefreshes(ctx context.Context, w Worker, budget *types.Currency, renterAddress types.Address, toRefresh []contractInfo) ([]api.ContractMetadata, []types.FileContractID, error) {
 	ctx, span := tracing.Tracer.Start(ctx, "runContractRefreshes")
 	defer span.End()
 
 	refreshed := make([]api.ContractMetadata, 0, len(toRefresh))
+	discarded := make([]types.FileContractID, 0, len(toRefresh))
```

Comment:
Do we treat contracts that failed to refresh due to unexpected reasons differently than the ones we call discarded because of the collateral? Don't we ignore both the same?

Reply:
Hm yeah that's silly. I used to archive them but after talking to you figured out that's not what we want either. We definitely need the two contract sets soon, I'll remove this "discarded" stuff.